code/modules/mob/living/carbon/carbon.dm

```diff
@@ -455,7 +455,7 @@
 		return 0
 	return ..()
 
-/mob/living/carbon/proc/vomit(lost_nutrition = 10, blood = FALSE, stun = TRUE, distance = 1, message = TRUE, toxic = FALSE, harm = TRUE, force = FALSE, purge = FALSE)
+/mob/living/carbon/proc/vomit(lost_nutrition = 10, blood = FALSE, stun = TRUE, distance = 1, message = TRUE, vomit_type = VOMIT_TOXIC, harm = TRUE, force = FALSE, purge_ratio = 0.1)
```

Comment:
0.1/0.67 serve special purposes/have a meaning behind them/should be used elsewhere right? is there any reason not to make them defines here?

Reply:
I could see defining 0.1 since it's the default, but leaving 0.67 not defined is kind of intentional, so future coders don't just think that it's either one or the other, and are encouraged to consider the value they want to use.

Reply:
Is 0.67 standard in some way? cause I see it a lot here, but it's a rather precise number. Have I misunderstood something?

Reply:
Previously it was a boolean that switched between the two values, probably coded that way because there was only one 'extra purge' vomit type at the time. I didn't want to rebalance things so i kept them as they were, but they could potentially be anything.

Reply:
my main concern is establishing standards, cause otherwise someone is gonna come along, go "I want a purge", and set it to 1, which doesn't seem like what we want.

Reply:
My concern was that someone would come along, see that there's VOMIT_PURGE_LOW and VOMIT_PURGE_HIGH, and only use those two instead of deciding the value for themselves.
That said, this is a pretty niche mechanic, so it doesn't matter much either way. If you prefer i'll set up the defines for all the existing values.

Reply:
that's good, it's not too big a deal